source/extensions/http/stateful_session/cookie/cookie.h

```diff
@@ -49,6 +53,10 @@ class CookieBasedSessionStateFactory : public Envoy::Http::SessionStateFactory {
     return !address.empty() ? absl::make_optional(std::move(address)) : absl::nullopt;
   }
 
+  bool requestPathMatch(absl::string_view request_path) const {
+    return absl::StartsWith(request_path, path_);
```

Comment:
Is this sufficient? A request path of `/foobar` will match the cookie path of `/foo` here, which seems incorrect.